### PR TITLE
Batching support

### PR DESCRIPTION
**Pull request by fmterrorf**

### CODE REVIEW

File: lib/commanded/event/handler.ex [hunk 1]
```diff
@@ -933,12 +1080,98 @@ defmodule Commanded.Event.Handler do
...
+    enriched_events =
+      Enum.map(events, fn e = %RecordedEvent{data: data} ->
+        {data, enrich_metadata(e, state)}
+      end)
+
+    try do
+      case handler_module.handle_batch(enriched_events) do
+        :ok ->
+          telemetry_stop(start_time, telemetry_metadata, :batch)
+          confirm_receipt(events, state)
```
Comment (seth) [inline lib/commanded/event/handler.ex:1137]: We pass `enriched_events` to the handler callback, but the unenriched list for error handling. Would it make sense to use the enriched_events here and below?

File: lib/commanded/event/handler.ex [hunk 2]
```diff
@@ -1022,21 +1264,28 @@ defmodule Commanded.Event.Handler do
...
 
         :timer.sleep(delay)
 
-        handle_event(failed_event, context, state)
+        retry_fun.(context, state)
 
       :skip ->
         # Skip the failed event by confirming receipt
         Logger.info(describe(state) <> " is skipping event")
+        state = confirm_receipt([maybe_failed_event], state)
```
Comment (seth) [inline lib/commanded/event/handler.ex:1280]: We call `confirm_receipt` in a few spots where we have just a single event. How about normalizing to lists in confirm_receipt so callers don't have to do the wrapping?

File: lib/commanded/event/handler.ex [hunk 3]
```diff
@@ -1022,21 +1264,28 @@ defmodule Commanded.Event.Handler do
...
 
-        confirm_receipt(failed_event, state)
+        if state.handler_callback == :batch do
+          %FailureContext{context: context} = failure_context
+          context = Map.put(context, :failure_action, :skip)
+          retry_fun.(context, state)
```
Comment (seth) [inline lib/commanded/event/handler.ex:1285]: Why do we retry if in batch mode here?

Comment (fmterrorf) [reply lib/commanded/event/handler.ex:1293]: I dont think we should retry here

File: lib/commanded/event/handler.ex (hunk 1 above)
Comment (fmterrorf) [reply lib/commanded/event/handler.ex]: I think I'm going to keep this as it is for now since this is also how handle_events behaves

**Pull request merged**